src/arrayset.rs

```diff
@@ -26,266 +32,274 @@ impl fmt::Display for InsertError {
+    A::CAPACITY
+  }
+
+  /// Returns the number of elements in the set
+  #[inline]
+  pub fn len(&self) -> usize {
+    self.len.into()
+  }
+
+  /// Returns true when the set contains no elements
+  #[inline]
+  pub fn is_empty(&self) -> bool {
+    self.len == 0.into()
+  }
+
+  /// Removes all elements
+  #[inline]
+  pub fn clear(&mut self) {
+    self.len = 0.into();
+  }
+
+  /// Iterate over all contents
+  #[inline]
+  pub fn iter(&self) -> Iter<A::Item> {
+    Iter { a: self.arr.as_slice(), i: 0 }
+  }
 }
 
 impl<A: Array, L> ArraySet<A, L>
 where
-    L: Copy + PartialOrd + AddAssign + SubAssign + From<u8> + Into<usize>,
+  L: Copy + PartialOrd + AddAssign + SubAssign + From<u8> + Into<usize>,
 {
-    /// Check whether the set contains `elt`
-    #[inline]
-    pub fn contains<Q: Eq + ?Sized>(&self, elt: &Q) -> bool where A::Item: Borrow<Q> {
-        self.get(elt).is_some()
-    }
+  /// Check whether the set contains `elt`
+  #[inline]
+  pub fn contains<Q: Eq + ?Sized>(&self, elt: &Q) -> bool
+  where
+    A::Item: Borrow<Q>,
+  {
+    self.get(elt).is_some()
+  }
 
-    /// Get a reference to a contained item matching `elt`
-    pub fn get<Q: Eq + ?Sized>(&self, elt: &Q) -> Option<&A::Item> where
-    A::Item: Borrow<Q> {
-        let len: usize = self.len.into();
-        let arr = self.arr.as_slice();
-        for i in 0..len {
-            if arr[i].borrow() == elt {
-                return Some(&arr[i]);
-            }
-        }
-        None
+  /// Get a reference to a contained item matching `elt`
+  pub fn get<Q: Eq + ?Sized>(&self, elt: &Q) -> Option<&A::Item>
+  where
+    A::Item: Borrow<Q>,
+  {
+    let len: usize = self.len.into();
+    let arr = self.arr.as_slice();
+    for i in 0..len {
+      if arr[i].borrow() == elt {
+        return Some(&arr[i]);
+      }
     }
+    None
+  }
 
-    /// Remove an item matching `elt`, if any
-    pub fn remove<Q: Eq + ?Sized>(&mut self, elt: &Q) -> Option<A::Item> where
-    A::Item: Borrow<Q> {
-        let len: usize = self.len.into();
-        let arr = self.arr.as_slice_mut();
-        for i in 0..len {
-            if arr[i].borrow() == elt {
-                let l1 = len - 1;
-                if i < l1 {
-                    arr.swap(i, l1);
-                }
-                self.len -= L::from(1);
-                let mut r = Default::default();
-                swap(&mut arr[l1], &mut r);
-                return Some(r);
-            }
+  /// Remove an item matching `elt`, if any
+  pub fn remove<Q: Eq + ?Sized>(&mut self, elt: &Q) -> Option<A::Item>
+  where
+    A::Item: Borrow<Q>,
+  {
+    let len: usize = self.len.into();
+    let arr = self.arr.as_slice_mut();
+    for i in 0..len {
+      if arr[i].borrow() == elt {
+        let l1 = len - 1;
+        if i < l1 {
+          arr.swap(i, l1);
         }
-        None
+        self.len -= L::from(1);
+        let mut r = Default::default();
+        swap(&mut arr[l1], &mut r);
+        return Some(r);
```

Comment:
Since it's experimental, may as well bypass the min-rust version :wink: 

```suggestion
        return Some(std::mem::take(&mut arr[l1]));
```

But otherwise, `replace` is still better

```suggestion
        return Some(std::mem::replace(&mut arr[l1], <A::Item>::default()));
```

Reply:
Actually I think we have a custom `take` in our lib.rs to support taking even with our older version.